### PR TITLE
#0: Add the dim 0 support repeat backward

### PR DESCRIPTION
**Pull request by ruthreshx**
Added the dim 0 support for  repeat backward.
yet to add the support for dim 1.

### CODE REVIEW

**Comment by ruthreshx** [comment]
Hi,
With the help of a [link](https://github.com/pytorch/pytorch/blob/3cd6a21e8f78683371220bb9a95acac6034d4724/torch/csrc/autograd/FunctionsManual.cpp#L1668) I have derived the repeat backward formula in python & it is working as we expected in colab for the various inputs.

The bottleneck is, the intermediate grad_size and sum_dims are incompatible for the TT support. It requires >4D dims to reshape and sum accordingly.

Due to this we were unable to add the entire support, but we were able to add the partial support for the dim 0 & 1, only when N and C is 1 and the repeat is of any number for such dim, other should be 1.

For the dim 2 & 3, it requires the shape should TILE, rather our one of the constraints is, the respective dim should be 1.

Ex: 
```
Input shape:
 [1, 1, 32, 32], [1,1, 320, 320] (N & C should be one)
           Repeats: 
 [12, 1, 1, 1], [6, 1, 1, 1], [1, 24, 1, 1], [1, 3, 1, 1]]  (repeat is of any number for such dim, other should be 1).

           For dim 0, Input is [1,1, 32, 32] the repeats for such dim should be any number.
		[32, 1, 1, 1], [2, 1, 1, 1], [6, 1, 1, 1]
           
	For dim 1, Input is [1,1, 64, 64] the repeats for such dim should be any number.
		[1, 24, 1, 1], [1, 30, 1, 1], [1, 14, 1, 1]
```

**Comment by tarafdarTT** [comment]
@ruthreshx this is awesome ty! This LGTM, I see you have added support at the tt_lib level. In a future PR we could look at adding this at the ttnn level as well.

**Review by tarafdarTT — APPROVED** [review]

File: tt_eager/tt_dnn/op_library/backward/backward_ops.cpp [hunk 1]
```diff
@@ -570,6 +572,43 @@ std::vector<Tensor> exp2_bw(const Tensor& grad, const Tensor& input, const Memor
 }
 
 
+// Repeat Backward
+std::vector<Tensor> _repeat_bw(const Tensor& grad, const Tensor& input, const Shape& shape, const MemoryConfig& output_mem_config) {
+    std::vector<Tensor> grad_tensor;
+    auto shape_wh = input.shape();
+    TT_ASSERT( shape_wh[0] == 1 && "input shape[0] should be 1");
```
Comment (tt-aho) [inline tt_eager/tt_dnn/op_library/backward/backward_ops.cpp:579]: All the TT_ASSERT should be switched to TT_FATAL here

**Comment by ruthreshx** [comment]
Fast dispatch: https://github.com/tenstorrent-metal/tt-metal/actions/runs/8453688540
Slow dispatch: https://github.com/tenstorrent-metal/tt-metal/actions/runs/8453708409

**Pull request merged**